src/fs-gen/src/loader/download.rs

```diff
@@ -0,0 +1,153 @@
+use reqwest::blocking::Client;
+use std::fs::create_dir_all;
+use std::path::{Path, PathBuf};
+use tracing::{debug, info};
+use anyhow::{Context, Result};
+use serde_json::Value;
+use crate::loader::errors::ImageLoaderError;
+use crate::loader::utils::{get_docker_download_token, unpack_tarball};
+
+pub(crate) fn download_image_fs(
+    image_name: &str,
+    architecture: &str,
+    output_file: PathBuf,
+) -> Result<Vec<PathBuf>, ImageLoaderError> {
+    info!("Downloading image...");
+
+    // Get image's name and tag
+    let image_and_tag: Vec<&str> = image_name.split(':').collect();
+    let image_name = image_and_tag[0];
+    let tag = if image_and_tag.len() < 2 { "latest" } else { image_and_tag[1] };
+
+    // Get download token and download manifest
+    let client = Client::new();
+    let token = &get_docker_download_token(&client, image_name)?;
+    let manifest = download_manifest(&client, token, image_name, tag)
+        .map_err(|e| ImageLoaderError::Error { source: e })?;
+
+
+    if let Some(layers) = manifest["layers"].as_array() {
```

Comment:
Could you please add a warning to the user in case we get directly the layers manifest and not the 'fat' manifest? Since the user can now pass the desired architecture in arguments, we should alert them if the image they use is non-multi-arch. Maybe something like: "WARNING: <image-name:tag> is not a multi-platform image, it is not guaranteed to work correctly on the <arch> architecture".